src/ui/util/TableConfig.ts

```diff
@@ -0,0 +1,93 @@
+import { idb } from "../../worker/db";
+import getCols, { LeagueVars, MetaCol } from "./columns/getCols";
+import { cloneDeep, uniq } from "lodash-es";
+import { g } from "../../worker/util";
+
+export class TableConfig {
+	public fallback: string[];
+	public columns: MetaCol[];
+	public tableName: string;
+	public vars: LeagueVars | object;
+
+	public statsNeeded: string[] = [];
+	public ratingsNeeded: string[] = [];
+	public attrsNeeded: string[] = ["pid"];
+
+	constructor(
+		tableName: string,
+		fallback: string[],
+		columns: MetaCol[] = [],
+		vars?: LeagueVars,
+	) {
+		this.tableName = tableName;
+		this.fallback = fallback;
+		this.columns = columns;
+		if (vars) this.vars = vars;
+	}
+
+	addColumn(column: MetaCol, pos: number) {
+		const colIndex = this.columns.findIndex(c => c.key === column.key);
+		if (colIndex !== -1) {
+			Object.assign(this.columns[colIndex], column);
+		} else {
+			this.columns.splice(pos, 0, column);
+		}
+	}
+
+	updateColumn(column: Partial<MetaCol>, key: string) {
+		const colIndex = this.columns.findIndex(c => c.key === key);
+		if (colIndex !== -1) {
+			Object.assign(this.columns[colIndex], column);
+		}
+	}
+
+	static unserialize(_config: TableConfig) {
+		const serialized = cloneDeep(_config);
```

Comment:
Use helpers.deepCopy instead.

Reply:
doing this broke things while building but will try again after doing this:
https://github.com/zengm-games/zengm/pull/406#discussion_r793881299